101-app-service-create-managed-certificate-apex/prereqs/prereq.azuredeploy.json

```diff
@@ -0,0 +1,80 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "domainName": {
+        "type": "string",
+        "metadata" : {
+                    "description":"Type your domain name. Ex.: contoso.com"
+        }
+        },
+        "websiteName": {
+            "type": "string",
+            "metadata" : {
+            "description":"Web App name that will host the Custom Domain. This name should be unique in Azure."
+        }
+        }
+    },
+    "variables": {
+       "certificateName": "[toLower(concat(parameters('domainName')))]",
+       "sku": "s1",
+       "workerSize": "0",
+       "AppServicePlanName": "[concat(toLower(parameters('websiteName')), 'ASP')]",
+    },
+  "resources": [
+   {
+      "apiVersion": "2018-02-01",
+      "name": "[variables('AppServicePlanName')]",
+      "type": "Microsoft.Web/serverfarms",
+      "location": "[resourceGroup().location]",
+      "sku": {
+        "name": "[variables('sku')]",
+        "capacity": "[variables('workerSize')]"
+      },
+      "properties": {
+        "name": "[variables('AppServicePlanName')]"
+      }
+   },
+   {
+      "apiVersion": "2018-02-01",
+      "name": "[parameters('websiteName')]",
+      "type": "Microsoft.Web/sites",
+      "location": "[resourceGroup().location]",
+        "properties": {
+        "serverFarmId": "[variables('AppServicePlanName')]"
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/serverfarms', variables('AppServicePlanName'))]"
+      ]
+   },
+       {
+      "apiVersion": "2018-11-01",
+      "name": "[concat(parameters('websiteName'), '/', parameters('domainName'))]",
+      "type": "Microsoft.Web/sites/hostNameBindings",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "sslState": "SniEnabled",
+        "thumbprint": "[reference(resourceId('Microsoft.Web/certificates', variables('certificateName'))).Thumbprint]"
```

Comment:
This resource doesn't exist in the template and as a prereq does not exist at time of deployment...  is this supposed to be in the main template?

Reply:
Fixed this part, but the hostnamebind will need to have a A DNS record pointing from the root domain to the web app just created. If the account has access to the DNS zone, we can automate this part, or validation will need to be make manually